woocommerce/woocommerce
# Reduce flakiness of `create-variable-product` and `create-coupon` specs

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR is a quick, short-term fix for the recent flakiness in the `create-variable-product` spec that started since https://github.com/woocommerce/woocommerce/pull/38288 was merged.

The fix includes:
- increasing the default test timeout, and the test timeout for tests within the `create-variable-product` spec
- reduce flakiness of `create-coupon` spec by waiting for the auto-save to complete before continuing with the rest of the steps
- added explicit waits in certain areas in those 2 specs
- replace the deprecated `page` methods in some areas within these two specs. All the other deprecated page methods will be replaced in a separate PR.

A longer term solution to make  `create-variable-product` more stable is already being made in a separate PR.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the [Run tests against PR](https://github.com/woocommerce/woocommerce/actions/runs/4991974584?pr=38321) check of this PR.
2. Verify that the "Run E2E tests" job passes.
3. Rerun the "Run E2E tests" job a few more times to see if it still fails, or if it consistently passes.
